src/__tests__/utils.test.ts

```diff
@@ -327,25 +327,45 @@ describe('utils', () => {
   });
 
   describe('getDefaultValues', () => {
-    test('static values', () => {
+    test('static values', async () => {
       const defaults: DefaultsOption<TestDocument> = {
         bar: 1,
         foo: 'test',
       };
 
-      const result = getDefaultValues(defaults);
+      const result = await getDefaultValues(defaults);
       expectType<Partial<TestDocument>>(result);
       expect(result).toStrictEqual(defaults);
     });
 
-    test('dynamic values', () => {
+    test('dynamic values', async () => {
       const defaults: DefaultsOption<TestDocument> = () => ({
         bar: 1,
         foo: 'test',
         ham: new Date(),
       });
 
-      const result = getDefaultValues(defaults);
+      const result = await getDefaultValues(defaults);
+      expectType<Partial<TestDocument>>(result);
+      expect(result.ham instanceof Date).toBeTruthy();
+    });
+
+    test('dynamic values with async function', async () => {
+      function getDateAsync(): Promise<Date> {
+        return new Promise((resolve) => {
+          setTimeout(() => {
+            resolve(new Date());
+          }, 2000);
```

Comment:
Please reduce this delay to something short like `10`. We shouldn't make unit tests slower due to testing async methods.

Reply:
done!